Sources/StreamChat/Models/Channel+Sorting.swift

```diff
@@ -0,0 +1,166 @@
+        channels.sorted { lhs, rhs in
+            for sort in sorting {
+                let stringKeyPath = sort.key.rawValue
+                let lhsValue = lhs[dynamicMember: stringKeyPath] ?? ""
+                let rhsValue = rhs[dynamicMember: stringKeyPath] ?? ""
+
+                if sort.isAscending, lhsValue < rhsValue {
+                    return true
+                } else if !sort.isAscending, lhsValue > rhsValue {
+                    return true
+                } else {
+                    continue
+                }
+            }
+
+            return false
+        }
+    }
+
+    struct SortValue<T> {
+        let keyPath: PartialKeyPath<T>
+        let isAscending: Bool
+    }
+
+    func sortWithKeyPath(with sorting: [SortValue<ChatChannel>]) -> [ChatChannel] {
+        func evaluate(lhs: Any?, rhs: Any?, isAscending: Bool) -> Bool {
+            if lhs == nil, rhs != nil, !isAscending {
+                return true
+            } else if lhs != nil, rhs == nil, isAscending {
+                return true
+            }
+
+            if let lString = lhs as? String, let rString = rhs as? String {
+                return isAscending ? lString < rString : lString > rString
+            } else if let lInt = lhs as? Int, let rInt = rhs as? Int {
+                return isAscending ? lInt < rInt : lInt > rInt
+            } else if let lDouble = lhs as? Double, let rDouble = rhs as? Double {
+                return isAscending ? lDouble < rDouble : lDouble > rDouble
+            } else if let lDate = lhs as? Date, let rDate = rhs as? Date {
+                return isAscending ? lDate < rDate : lDate > rDate
+            } else if let lBool = lhs as? Bool, let rBool = rhs as? Bool {
+                return isAscending ? !lBool && rBool : lBool && !rBool
+            }
+
+            return false
+        }
+
+        return channels.sorted { lhs, rhs in
+            for sort in sorting {
+                let lhsValue = lhs[keyPath: sort.keyPath]
+                let rhsValue = rhs[keyPath: sort.keyPath]
+
+                if sort.isAscending, evaluate(lhs: lhs, rhs: rhs, isAscending: sort.isAscending) {
+                    return true
+                } else if !sort.isAscending, evaluate(lhs: lhs, rhs: rhs, isAscending: sort.isAscending) {
+                    return true
+                } else {
+                    continue
+                }
+            }
+
+            return false
+        }
+    }
+}
+
+extension ChatChannel {
+    // This method is meant to only be used for sorting functionality.
+    subscript(dynamicMember member: String) -> String? {
+        switch member {
+        case "cid":
+            return cid.rawValue
+        case "name":
+            return name
+        case "type":
+            return type.rawValue
+        case "lastMessageAt":
+            return lastMessageAt?.sortValue
+        case "createdBy":
+            return createdBy?.id
```

Comment:
The keys should a actually be the ones from the backend 
```suggestion
        case "created_at":
            return createdBy?.id
```